src/modules/Host.rb

```diff
@@ -207,14 +207,13 @@ def Import(settings)
     # Dump the Hosts settings to a map, for autoinstallation use.
     # @return autoinstallation settings
     def Export
-      if Ops.greater_than(Builtins.size(@hosts), 0)
-        # Filter out IPs with empty hostname (so that valid autoyast
-        # profile is created)(#335120)
-        @hosts = Builtins.filter(@hosts) { |_ip, names| names != [] }
-        return { "hosts" => Builtins.eval(@hosts) }
-      else
-        return {}
-      end
+      return {} if hosts.empty?
+
+      # Filter out IPs with empty hostname (so that valid autoyast
+      # profile is created)(#335120)
+      hosts.keep_if { |_, names| !names.empty? }
```

Comment:
this creates quite nasty side-effect which result that Export modify hosts element, which looks wrong from my POV.